cpp/ql/src/semmle/code/cpp/models/implementations/Memset.qll

```diff
@@ -0,0 +1,38 @@
+import semmle.code.cpp.Function
+import semmle.code.cpp.models.interfaces.ArrayFunction
+import semmle.code.cpp.models.interfaces.DataFlow
+import semmle.code.cpp.models.interfaces.Taint
+
+/**
+ * The standard function `memset` and its assorted variants
+ */
+class MemsetFunction extends ArrayFunction, DataFlowFunction, TaintFunction {
+  MemsetFunction() {
+    this instanceof TopLevelFunction and
+    (
+      this.hasName("memset") or
+      this.hasName("__builtin_memset") or
```

Comment:
The modern way to match these function names is to use `hasGlobalName` and the multi-argument `hasQualifiedName` predicates. The even more modern way (#1585) is not merged yet, unfortunately, so you'll have to add `hasQualifiedName("std", "memset")` to make sure you also match `std::memset`.

Reply:
Done in https://github.com/Semmle/ql/pull/2027.